### PR TITLE
Recursos adicionales para base de datos

### PR DESCRIPTION
**Pull request by RicardoPedreros**

### CODE REVIEW

File: database/src/resources/EnsayoConsultas.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+        // Se guarda el resultado de la consulta en una arreglo de Objetos
+        // Nota: Siempre va a tener dos posiciones.
+        Object[] objeto= aQuery.read_DB("SELECT * FROM users;");
+
+        // En la posición 0 va a contener el tipo de resultado, solo existen
+        // dos tipos de resultados, "Error" cuando algo falló en el query,
+        // o "QueryResult" cuando la consulta arrojó al menos un registro.
+        String tipo_resultado = (String) objeto[0];
+        System.out.println("\n El tipo de resultado que arrojo la consulta fue un " + tipo_resultado + "\n");
```
Comment (johanf219) [inline database/src/resources/EnsayoConsultas.java:18]: is this println necessary?

File: database/src/resources/EnsayoConsultas.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+
```
Comment (johanf219) [inline database/src/resources/EnsayoConsultas.java]: remove space

File: database/src/resources/EnsayoConsultas.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        // En caso que el resultado sea "QueryResult", el manejo de los registros
+        // sería el siguiente:
+        Vector<String[]> misRegistros = (Vector<String[]>) objeto[1];
+
+        // Para acceder al primer registro utilizamos el método get
+        String[] tupla_Uno = misRegistros.get(0);
+
+        // Este arreglo de Strings contiene cada atributo del registro,
+        // en este caso la consulta no nos dice cuantos atributos
+        // traerá (SELECT * FROM empleados), si deseamos saber cuantos
```
Comment (johanf219) [inline database/src/resources/EnsayoConsultas.java:32]: empleados or users? read line 15

File: database/src/resources/EnsayoConsultas.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        for(int i = 0; i < cantidadRegistros; i++){
+            tupla = misRegistros.get(i);
+            System.out.println("Atributos de la tupla "+ (i+1) + ":");
+
+            for(int j = 0; j < numeroAtributos; j++){
+                System.out.println("  " + (j+1) + ": " + tupla[j] );
+            }
+        }
+
+
```
Comment (johanf219) [inline database/src/resources/EnsayoConsultas.java]: remove spaces

File: database/src/resources/EnsayoConsultas.java [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+        // Si el resultado es un error, el vector tendría UNA SOLA POSICIÓN
+        // con un arreglo de Strings con DOS posiciones.
+        // EJEMPLO: Vector<String[]> miError =  (Vector<String[]>) objeto[1];
+        // El vector miError, como dije anteriormente, solo tendría una posicion
+        // para acceder a ella
+
```
Comment (johanf219) [inline database/src/resources/EnsayoConsultas.java]: remove space

File: database/src/resources/EnsayoConsultas.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
```
Comment (johanf219) [inline database/src/resources/EnsayoConsultas.java]: remove space

File: database/src/resources/RAJA_DB.sql [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+-- ESTA TABLA ES PARA LLEVAR EL CONTROL DE QUE SE ESTÁ UTILIZANDO LA VERSIÓN
+-- ADECUADA CON LA APLICACIÓN Y EVITAR CONFLICTOS ENTRE ELLAS.
+CREATE TABLE DB_version (
+	version INT,
+	CONSTRAINT pk_db_version PRIMARY KEY (version)
+);
+INSERT INTO DB_version VALUES(4);
+
+
```
Comment (johanf219) [inline database/src/resources/RAJA_DB.sql]: remove space

File: database/src/resources/RAJA_DB.sql [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+	userID INT NOT NULL,
+	numberForPay VARCHAR(28) NOT NULL,
+	CONSTRAINT pk_comsumption PRIMARY KEY (id),
+	CONSTRAINT fk_consumptionline FOREIGN KEY (lineNumber) REFERENCES Lines(number) ON UPDATE CASCADE ON DELETE RESTRICT,
+	CONSTRAINT fk_billuser FOREIGN KEY (userID) REFERENCES Users(id) ON UPDATE CASCADE ON DELETE RESTRICT
+);
+
+-- ****************VISTAS*********************
+
+
```
Comment (johanf219) [inline database/src/resources/RAJA_DB.sql]: remove space

File: database/src/resources/RAJA_DB.sql [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+-- ******************PROCEDIMIENTOS ALMACENADOS**********************
+
+
+
```
Comment (johanf219) [inline database/src/resources/RAJA_DB.sql]: remove space

File: database/src/resources/RAJA_DB.sql [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+-- *******************INSERTS PERMANENTES*******************
+INSERT INTO Roles VALUES
+(1, 'Administrador','Descripcion del administrador'),
+(2, 'Gerente','Descripcion del gerente'),
+(3, 'Operador','Descripcion del operador');
+
+
```
Comment (johanf219) [inline database/src/resources/RAJA_DB.sql]: remove space

File: database/src/resources/RAJA_DB.sql [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+INSERT INTO Customer(id, name, type, email)  VALUES(1007151295, 'Andrés Viáfara', 'Natural','dianbovi@hotmail.com');
+INSERT INTO Lines VALUES('3107356146', 1007151295, 4, 456, true);
+INSERT INTO Bill VALUES(2, 10000, 800, 1050, 5000, 200, 8060, 1024, 0, 0, 'Davivienda', 65000, '3107356146', '2020/03/30', 123,'1007151295310735614620200330');
+
+INSERT INTO Customer(id, name, type, address, city)  VALUES(1006106575, 'Pablo Esteban', 'Natural','Noc','Piendamó');
+INSERT INTO Lines VALUES('3217219953', 1006106575, 5, 456, true);
+INSERT INTO Bill VALUES(3, 25000,8000, 4000,7000,400,10060,2048, 60, 500, 'Davivienda', 100000, '3217219953', '2020/03/30',123,'1006106575321721995320200330');
+
+
```
Comment (johanf219) [inline database/src/resources/RAJA_DB.sql]: remove space

File: database/src/resources/RAJA_DB.sql [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+INSERT INTO Bill VALUES(4, 110, 210, 60, 160, 410, 170, 0, 0, 0, 'Bancolombia', 30900, '3219234114', '2020/02/28',123,'1007151952321923411420200228');
+
+
+
+
```
Comment (johanf219) [inline database/src/resources/RAJA_DB.sql]: remove spaces

File: database/src/resources/RAJA_DB.sql [hunk 13]
```diff
@@ -0,0 +1,168 @@
...
+INSERT INTO Bill VALUES(4, 110, 210, 60, 160, 410, 170, 0, 0, 0, 'Bancolombia', 30900, '3219234114', '2020/02/28',123,'1007151952321923411420200228');
+
+
+
+	
+-- **********************************************************
+-- **********************************************************
+
+-- *****************QUERY'S DEL SISTEMA**********************
+
```
Comment (johanf219) [inline database/src/resources/RAJA_DB.sql]: remove space

File: database/src/resources/EnsayoConsultas.java [hunk 14]
```diff
@@ -0,0 +1,78 @@
+package resources;
+
+import java.util.Vector;
+import DB_Connection.DBConnection;
+
+public class EnsayoConsultas {
+
+    public static void main (String args[]){
+
```
Comment (johanf219) [inline database/src/resources/EnsayoConsultas.java]: remove space

File: database/src/resources/EnsayoConsultas.java [hunk 15]
```diff
@@ -0,0 +1,78 @@
+package resources;
+
+import java.util.Vector;
+import DB_Connection.DBConnection;
+
+public class EnsayoConsultas {
+
```
Comment (johanf219) [inline database/src/resources/EnsayoConsultas.java]: remove space

File: database/src/resources/EnsayoConsultas.java (hunk 1 above)
Comment (johanf219) [reply database/src/resources/EnsayoConsultas.java:18]: why are we using println? i dont think we should show this information.
i not sure about this file
the functionality is ok but you talk about condictionals but i dont see anyone (read line 59 "si queremos acceder") this functionalities should be functions

**Review by johanf219 — CHANGES_REQUESTED** [review]

**Review by HttpKiwi — APPROVED** [review]

**Pull request merged**